tests/metrics/test_nlp_metrics.py

```diff
@@ -200,3 +201,22 @@ def test_perplexity():
     perplexity = perplexity_metric.compute()
 
     assert torch.equal(torch.exp(ce), perplexity)
+
+
+def test_in_context_learning_lm_accuracy(tiny_gpt2_tokenizer):
+    pad = tiny_gpt2_tokenizer.eos_token_id
+    input = tiny_gpt2_tokenizer('The dog is')['input_ids'] + tiny_gpt2_tokenizer(' furry')['input_ids']
+
+    start = len(tiny_gpt2_tokenizer('The dog is')['input_ids'])
+    end = len(tiny_gpt2_tokenizer('The dog is')['input_ids']) + len(tiny_gpt2_tokenizer(' furry')['input_ids'])
+    cont_idxs = torch.tensor(list(range(start, end)))
+
+    input = torch.tensor([input + [pad] * (2048 - len(input))])
+
+    batch = {'continuation_indices': [cont_idxs], 'labels': input, 'input_ids': input}
+    logits = torch.nn.functional.one_hot(torch.roll(input, shifts=-1), num_classes=pad + 1)
+
+    metric = InContextLearningLMAccuracy()
```

Comment:
Maybe add another one that tests something other than 1? Like an artificial logits that get you 50% accuracy. and maybe have a batch size greater than 1. just to cover the scenarios a little bit more.